### ISSUE TITLE
Missing Schema Compare functionality

### ISSUE THREAD

**Issue by etherwizard** [microsoft/vscode-mssql#1053]
- MSSQL Extension Version: 1.3.0
- VSCode Version: 1.19.2
- OS Version: macOS 10.13.2, Docker 17.12.0-ce-mac47

Is Schema Compare on the roadmap for the **mssql** extension? It is becoming extremely tedious to track and deploy database updates between development, test, and production environments. A visual tool similar to Schema Compare as seen in SQL Server Data Tools for Visual Studio would be a very welcome feature.

**Comment by joshgrib** [comment]
I would really like this, currently schema comparisons are the only reason I still need to open VS17, I prefer using VS Code much more

**Comment by vyrotek** [comment]
I find myself in the same position. I would really like to perform comparisons and generate deployment scripts from VSCode in the same way that I do in VS using SQL Server Database Projects.

**Comment by R-i-c-h-B** [comment]
Seems to be coming on for a year since this was suggested, does anyone know if there has been a solution.  

I see over on the Azure Data Studio that they seem to expect the VCode extensions to be handling the more complete Dev program that was provided by Visual Studio, but I'm not feeling it... Have I missed something?

**Comment by kburtram** [comment]
@R-i-c-h-B it's still the plan to add additional dev scenario to the VS Code extension, with schema compare being one of the top scenarios.  

We have a relatively small team working across dev, dba, and data engineering workloads (while supporting our flagship SSMS and SSDT product lines), so realistically I'm not sure when this will land.  We are introducing "Import Dacpac" in Azure Data Studio Dec release which brings DacFx to .Net Core, which is the main library that enables SQL Project and Schema Compare.  So at least this removes one of the main blockers for building this feature.

**Comment by R-i-c-h-B** [comment]
@kburtram  - understood.  

It's unfortunate that this seems a lesser importance than those projects, especially as they are so very firmly locked into the Windows space while so much effort has been put into allowing SQL Engine itself to compete on the linux market.  

Getting us to a place where we can have a solid and indeed market leading database development experience under linux would be very nice indeed :) - especially as so much of the implementation has been done albeit in the wrong place!

Big fan of what you are doing, and if you get this right I do believe it will become the go to for all platforms.  Build in extensibility to handle the core Azure and Apache data layers and other old fashioned rdbms like postgres/mysql under the same version selectable dev/deploy model and it becomes something quite extraordinary.

**Comment by BlakesHeaven** [comment]
+1 and a thumbs up on the original post.
Yes please - bring us a SQL Server Database Schema comparison extension, please.

**Comment by ozer-a1** [comment]
> Hi @kburtram !
> 
**paying a high license fee for VS IDE just to do schema comparisons**.


schema compare is available in VS Community Edition for free.

**Comment by lestergill** [comment]
As a Lead Data Engineer, I mostly develop in Visual Studio Code, for Python APIs, Spring microservices, using WSL, interacting with VM's, SQL development, and so on. 

Since this feature is in Visual Studio and Azure Data Studio already, I do not see a reason why it is not a feature in VS Code as well. I suspect that it is a strategic directive, as it may push more developers away from Azure Data Studio and towards Visual Studio Code instead. It is laborious to use Azure Data Studio for this, since the entirety of the rest of my stuff is elsewhere. I'm sure you can agree that Data Engineers have the burden of being cross-functional with wide skill sets, and it would be absolutely welcomed for you to unburden them by supplying a strongly used feature in the tools that they use, in addition to that feature existing in the silo of ADS.

**Comment by PeterFogh** [comment]
I agree with @lestergill - I would really appreciate is VScode got the "SQL Server Schema Compare" extension like Azure Data Studio has.

**Comment by besmirbz** [comment]
6 years later, we are still here. Sad days

**Comment by pheeca** [comment]
I would really appreciate this. 

+1 to op

**Comment by lausitzer** [comment]
Azure Data Studio will soon be retired and points us to vscode-mssql.  https://learn.microsoft.com/en-us/azure-data-studio/whats-happening-azure-data-studio

But without schema comparison and scripting, vscode-mssql isn't the solution. Please support this features.

**Comment by croblesm** [comment]
Hello everyone!

Thank you, @etherwizard and everyone else, for your patience. I just want to confirm what @jikuja mentioned earlier. We recently [published a blog post](https://devblogs.microsoft.com/azure-sql/mssql-vs-code-whats-new-next/#looking-ahead:-open-roadmap) announcing that our roadmap is now publicly available.

As highlighted in the post, we recognize that the **Schema Management** experience in VS Code is not yet complete. We're actively working on this and plan to address it in our April release, which will introduce both the **Schema Compare** and **Schema Designer** features in Public Preview.

I'll go ahead and close this issue now. If you're interested in learning more about what's coming, feel free to check out our public roadmap:  
https://aka.ms/vscode-mssql-roadmap
